src/components/SearchForm/SearchForm.jsx

```diff
@@ -0,0 +1,132 @@
+import { useState } from "react";
+import { useForm } from "react-hook-form";
+import { useHistory } from "react-router-dom";
+import PropTypes from "prop-types";
+import clsx from "clsx";
+import { Select } from "../Select/Select";
+import { DataList } from "../DataList/DataList";
+import { Checkbox } from "../Checkbox/Checkbox";
+import { SERVICES, CATEGORIES, CITIES, PETS } from "../../constants/options";
+import styles from "./search_form.module.scss";
+import { Button } from "../Button/Button";
+import searchButtonIcon from "../../assets/image/searchButton.png";
+
+export const SearchForm = ({ filterAnnouncements, setFilters }) => {
+  const [filterState, setFilterState] = useState([]);
+  const history = useHistory();
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+  } = useForm();
+
+  const createSearchParamsString = (data) => {
+    const params = new URLSearchParams();
+    Object.keys(data).forEach((key) => {
+      if (Array.isArray(data[key])) {
+        data[key].forEach((element) => {
+          params.set(key.toLowerCase(), element);
+        });
+      } else {
+        params.set(key.toLowerCase(), data[key]);
+      }
+    });
+    return params;
```

Comment:
The method is called `createSearchParamsString` so why you just don't return `params.toString()`?